shared/src/test/diff/qq/EffectfulLetInsertion.mls

```diff
@@ -0,0 +1,126 @@
+:NewDefs
+
+:NoJS
```

Comment:
Now that we have mutation, can we try to implement this one? (Using QQ pretty-printing for the codegen.) The `Scope` class would have to use some unsafe casts under the hood but could still present a type-safe interface. It would accumulate bindings registered using `bind`.

Reply:
Whoops, I realized that we'd need a bit more to actually do this, such as effect handlers or first-class `Var` values. Not in the scope of this PR; please ignore.